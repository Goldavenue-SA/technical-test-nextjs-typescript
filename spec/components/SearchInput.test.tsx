import { fireEvent, render, screen } from '@testing-library/react';
import {
  SearchInput,
  SearchInputProps,
} from '../../components/SearchInput/SearchInput';
import '../setupTests';

describe('Test Search Input Component', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  const searchInputProps: SearchInputProps = {
    id: 'someID',
    label: 'someLabel',
    placeholder: 'somePlaceHolder',
    onSearch: jest.fn(),
  };

  const setupComponent = () => {
    return render(<SearchInput {...searchInputProps} />);
  };

  it('renders correctly and matches the snapshot', () => {
    const { asFragment } = setupComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with the correct label and placeholder', () => {
    setupComponent();

    expect(screen.getByLabelText(searchInputProps.label)).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText(searchInputProps.placeholder)
    ).toBeInTheDocument();
  });

  it('calls onSearch with the correct query on input change', () => {
    setupComponent();

    const input = screen.getByPlaceholderText(searchInputProps.placeholder);

    fireEvent.change(input, { target: { value: 'Hello' } });

    expect(searchInputProps.onSearch).toHaveBeenCalledWith('Hello');

    fireEvent.change(input, { target: { value: 'Hello World' } });

    expect(searchInputProps.onSearch).toHaveBeenCalledWith('Hello World');
  });

  it('calls onSearch with an empty string when input is cleared', () => {
    setupComponent();

    const input = screen.getByPlaceholderText(searchInputProps.placeholder);

    fireEvent.change(input, { target: { value: 'Search Term' } });

    expect(searchInputProps.onSearch).toHaveBeenCalledWith('Search Term');

    fireEvent.change(input, { target: { value: '' } });

    expect(searchInputProps.onSearch).toHaveBeenCalledWith('');
  });
});
