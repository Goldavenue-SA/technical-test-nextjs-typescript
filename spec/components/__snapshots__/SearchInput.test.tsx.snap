// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search Input Component renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-3134fa0a-0 laGSnx"
  >
    <label
      class="sc-3134fa0a-2 hMoKSw"
      for="someID"
    >
      someLabel
    </label>
    <input
      aria-describedby="search-help-text"
      aria-label="Search input"
      class="sc-3134fa0a-1 eZYfBK"
      id="someID"
      placeholder="somePlaceHolder"
      type="search"
    />
  </div>
</DocumentFragment>
`;
